lib/ipam/interface.go

```diff
@@ -36,7 +36,7 @@ type Interface interface {
 	// which is useful for dataplanes that need to know the subnet (such as Windows).
 	//
 	// In case of error, returns the IPs allocated so far along with the error.
-	AutoAssign(ctx context.Context, args AutoAssignArgs) ([]cnet.IPNet, []cnet.IPNet, error)
+	AutoAssign(ctx context.Context, args AutoAssignArgs) ([]cnet.IPNet, []cnet.IPNet, *IPAMAssignmentInfo, *IPAMAssignmentInfo, error)
```

Comment:
I think it might be a good idea to include the assigned IP addresses as well into this. So your return values would be just `IPAMAssignmentInfo` (or whatever this will be called after it's merged with the return info) and `error`

Reply:
Hm, I see, it makes sense. Thanks, Matt! I guess we could call it `IPAMAssignment` and have it contain the []cnet.IPNet slice inside it. Is that name good?